platalea/preprocessing.py

```diff
@@ -7,40 +6,41 @@
 import json
 import logging
 from scipy.io.wavfile import read
-import os
 import numpy
+import platalea.config
+import pathlib
 
 
-def flickr8k_features():
-    config = dict(audio=dict(dir=CONFIG['flickr8k_root'],
-                             type='mfcc', delta=True, alpha=0.97, n_filters=40,
-                             window_size=0.025, frame_shift=0.010),
-                  image=dict(dir=CONFIG['flickr8k_root'],
-                             model='resnet'))
-    flickr8k_audio_features(config['audio'])
-    flickr8k_image_features(config['image'])
+_device = platalea.config.device()
+
+
+def flickr8k_features(dataset_path=platalea.config.args.data_root,
+                      audio_subdir=platalea.config.args.audio_subdir,
+                      images_subdir=platalea.config.args.image_subdir):
+    audio_config = dict(dataset_path=pathlib.Path(dataset_path), audio_subdir=audio_subdir, type='mfcc',
+                        delta=True, alpha=0.97, n_filters=40, window_size=0.025,
+                        frame_shift=0.010)
+    images_config = dict(dataset_path=pathlib.Path(dataset_path), images_subdir=images_subdir, model='resnet')
+    flickr8k_audio_features(audio_config)
+    flickr8k_image_features(images_config)
 
 
 def flickr8k_audio_features(config):
-    directory = os.path.join(config['dir'], 'flickr_audio/wavs')
-    wav2capt_fname = os.path.join(config['dir'], 'flickr_audio/wav2capt.txt')
-    files = [line.split()[0] for line in open(wav2capt_fname)]
-    paths = [os.path.join(directory, file) for file in files]
+    directory = config['dataset_path'] / config['audio_subdir']
+    files = [line.split()[0] for line in open(config['dataset_path'] / 'wav2capt.txt')]
```

Comment:
Same as before, wav2capt.txt is by default under flickr_audio folder
```suggestion
    files = [line.split()[0] for line in open(config['dataset_path'] / 'flickr_audio' / 'wav2capt.txt')]
```

Reply:
Same as before indeed.